internal/app/machined/pkg/system/runner/process/process.go

```diff
@@ -214,13 +214,17 @@ func (p *processRunner) build() (commandWrapper, error) {
 		return commandWrapper{}, err
 	}
 
-	go io.Copy(writer, pr) //nolint:errcheck
+	go func() {
+		defer pr.Close() //nolint:errcheck
```

Comment:
this was a bug, copying from `pr` to `writer` was not synchronized with closing `pr` - there is a chance that `pr` will be closed before copying is done